tests/integration/test_charm.py

```diff
@@ -17,33 +25,124 @@ async def juju_run(unit, cmd):
     return stdout
 
 
-async def test_build_and_deploy(ops_test: OpsTest):
+async def test_workload_online_default(ops_test: OpsTest, app: Application):
     assert ops_test.model
-    charm = await ops_test.build_charm(".")
-    await ops_test.model.deploy("postgresql-k8s")
-    await ops_test.model.deploy(charm)
-    await ops_test.model.wait_for_idle(raise_on_error=False)
-    await ops_test.model.add_relation(
-        "postgresql-k8s:db",
-        "mattermost-k8s",
+    mmost_unit = app.units[0]
+    action = await mmost_unit.run("unit-get private-address")
+    curl_output = await juju_run(
+        mmost_unit, "curl {}:8065".format(action.results["Stdout"].replace("\n", ""))
     )
-    # mypy doesn't see that ActiveStatus has a name
-    await ops_test.model.wait_for_idle(status=ActiveStatus.name, raise_on_error=False)  # type: ignore
+    assert "Mattermost" in curl_output
+
 
+async def test_s3_storage(
+    ops_test: OpsTest,
+    app: Application,
+    localstack_s3_config: dict,
+):
+    """
+    arrange: after charm deployed and openstack swift server ready.
+    act: update charm configuration for openstack object storage plugin.
+    assert: a file should be uploaded to the openstack server and be accessible through it.
+    """
 
-async def test_status(ops_test: OpsTest):
     assert ops_test.model
```

Comment:
We have this following standard, is it applicable here? https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#handling-typing-issues-with-python-libjuju

Reply:
Fixed